testing/simapp/app.go

```diff
@@ -938,6 +954,28 @@ func (app *SimApp) setupUpgradeHandlers() {
 			app.configurator,
 			app.appCodec,
 			app.IBCKeeper.ClientKeeper,
+			app.ConsensusParamsKeeper,
+			app.ParamsKeeper,
 		),
 	)
 }
+
+// setupUpgradeStoreLoaders sets all necessary store loaders required by upgrades.
+func (app *SimApp) setupUpgradeStoreLoaders() {
+	upgradeInfo, err := app.UpgradeKeeper.ReadUpgradeInfoFromDisk()
+	if err != nil {
+		tmos.Exit(fmt.Sprintf("failed to read upgrade info from disk %s", err))
+	}
+
+	if upgradeInfo.Name == v7.UpgradeName && !app.UpgradeKeeper.IsSkipHeight(upgradeInfo.Height) {
+		storeUpgrades := storetypes.StoreUpgrades{
+			Added: []string{
+				consensusparamtypes.StoreKey,
+				crisistypes.StoreKey,
```

Comment:
Adds store keys for `x/consensus` and `x/crisis`. 

The new `x/consensus` module maintains consensus parameters previously maintained by the base app subspace.
The `x/crisis` module previously existed but _without_ a store key. With the deprecation of `x/params`, the crisis module now uses its own dedicated store to maintain params.

Reply:
Looks like this logic is repeated for every new module. If we need to do it again for a new module, it'd probably make sense to turn into a helper function. Looks great to me as is

Reply:
Yeah I agree. We could possible adopt the osmosis [upgrade structure](https://github.com/osmosis-labs/osmosis/blob/main/app/upgrades/v15/constants.go#L18), but may require some refactoring of app.go.